templates/layout-slider.php

```diff
@@ -3,31 +3,36 @@
 
 $show_navigation = count($images) >= $nav_count_min;
 $slider_image_size = has_image_size('post-thumbnail') ? 'post-thumbnail' : 'large';
+
+$image_attr = [];
 ?>
 
 <div class="gallerya gallerya--slider">
   <ul class="js-gallerya-slider js-gallerya-lightbox">
+    <?php $image_attr = Plugin::lazyLoadIsActive() ? ['data-no-lazy' => '1'] : []; ?>
     <?php foreach ($images as $image):
       $caption = apply_filters('gallerya/image_caption', $image->post_excerpt, $image->ID);
     ?>
       <li>
         <figure class="gallerya__image">
           <a href="<?= wp_get_attachment_image_src($image->ID, apply_filters('gallerya/image_size_lightbox', 'large'))[0] ?>" <?= !empty($caption) ? 'data-sub-html="' . esc_attr($caption) . '"' : '' ?>>
-            <?= wp_get_attachment_image($image->ID, apply_filters('gallerya/image_size_slider', $slider_image_size)) ?>
+            <?= wp_get_attachment_image($image->ID, apply_filters('gallerya/image_size_slider', $slider_image_size), FALSE, $image_attr) ?>
           <?php if (!empty($caption)): ?>
             <figcaption class="gallerya__image__caption"><?= $caption ?></figcaption>
           <?php endif; ?>
           </a>
         </figure>
       </li>
+      <?php $image_attr = []; ?>
```

Comment:
Instead of clean the array after the `foreach` can we not add a condition to apply this just to the first image? In that way we can get the rid of this line as well of the line 31 that is duplicating what line 12 is already doing.

Reply:
1. Why are you resetting / emptying the variable after the first iteration of the foreach loop?

1. Why do we need to tell the lazyloading library to not apply lazy-loading to the first image in a slider in the first place?  Is this PR trying to fix (work around) a defect in a JS library?

Reply:
Flickity is unable to calculate correctly the height of the slider images if lazyload is enabled and those images cached (by wp-rocket). The workaround I found is to disable lazyload just for the first image of the slider.

For the thumbnails strip and grid-slider layout this is not enough, and lazyload has to be disabled for every image.